### DISCUSSION TITLE
Azure FunctionsでKeyVaultが利用できない

### DISCUSSION THREAD

**Discussion by RyogaCloud** [1ft-seabass/ms-azure-hackathon-league-2021#58] [Q&A] [answered]
https://servian.dev/accessing-azure-key-vault-from-python-functions-44d548b49b37
上記サイトを参考にしましたが、下図のように認証されません。
解決方法を教えていただきたいです。
![](https://user-images.githubusercontent.com/45579270/146659853-b9a327ab-469f-4c38-9aca-b4560dd20723.png)

**Comment by papeMK2** [comment, marked as answer]
## 対処方法

* `KeyVault の管理画面 > アクセスポリシー > アクセスポリシーの追加` に進む

![image](https://user-images.githubusercontent.com/1393824/146660551-73c01ac5-5b68-4b6a-aa43-838a669d7404.png)

* `テンプレートの構成` から権限テンプレートを選択

![image](https://user-images.githubusercontent.com/1393824/146660600-68e412c7-b7e9-4e7c-897a-ebe3817afd73.png)

* キー、シークレット、証明書について、それぞれ付与する権限を選択する

![image](https://user-images.githubusercontent.com/1393824/146660616-49b988b7-e432-4864-9812-66ba22f11598.png)

* `プリンシパルの選択` を選択、右に出たペインから、権限を付与したいアプリケーションを検索、選択

![image](https://user-images.githubusercontent.com/1393824/146660660-ba77ccb4-20c4-4937-99bd-d97e2525c918.png)

* 最後に追加ボタンで権限追加

## 注意点

* 権限を付与する場合は、最小限の権限に留めましょう。(アプリケーションなら取得権限のみ等)
